src/main/java/command/DeleteCommand.java

```diff
@@ -0,0 +1,43 @@
+package command;
+
+import data.DukeException;
+import data.Storage;
+import data.TaskList;
+import data.Ui;
+
+/**
+ * Command that deletes a Task from Tasklist when executed.
+ */
+public class DeleteCommand extends Command {
+    /**
+     * Index of the task in TaskList
+     */
+    private int taskNumber;
+
+    public DeleteCommand(int taskNumber) {
+        this.taskNumber = taskNumber;
+    }
+
+    /**
+     * Deletes a task from the Tasklist.
+     *
+     * @param tasks The list of tasks that a user has
+     * @param ui The ui that sends a message to the user once the task is deleted
+     * @param storage Saves the updated TaskList to disk
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (taskNumber > tasks.getSize() || taskNumber <= 0) {
+            throw new DukeException("Please insert a valid task.Task Number!");
```

Comment:
Hey not sure if "task.Task" is a clear instruction to user, people might not understand.

Reply:
The extra "task" seem to have been accidentally added in when i was refactoring the Task class, thanks for pointing it out! I've changed it to "Task" instead of "task.Task".